examples/solid/basic-typescript/src/index.tsx

```diff
@@ -97,13 +99,11 @@ const getPostById = async (id: number): Promise<Post> => {
 }
 
 function createPost(postId: number) {
-  return createQuery(
-    () => ['post', postId],
-    () => getPostById(postId),
-    {
-      enabled: !!postId,
-    },
-  )
+  return createQuery({
+    queryKey: () => ['post', postId],
+    queryFn: () => getPostById(postId),
+    enabled: !!postId,
+  })
```

Comment:
also here, I think this should be:

```
createQuery(() => ({ queryKey, queryFn }))
```

Reply:
It is my first time dealing with solid tbh, but I think the object syntax is working fine with the tests 

@ardeora is the object syntax fine? or should it be converted to an arrow function returning the object?

Reply:
Thanks for working on this! Looks great. Yeah solid query `v5` will need more changes here. With the removal of overloads. We don't need to have `QueryKey` as a function anymore and would be more like what Dominik mentioned above. For example,

```ts
createQuery({
    queryKey: () => ['post', postId],
    queryFn: () => getPostById(postId),
    enabled: !!postId,
  })
```

will become 

```ts
createQuery(() => {
    queryKey: ['post', postId],
    queryFn: () => getPostById(postId),
    enabled: !!postId,
  })
```

But this change is definitely wayyy more involved since we need to change the typings in `types.ts` to get in line with with `react-query`. It would also change how `createQuery` tracks the signal changes inside the function. This would also mean changing the tests to reflect the same. @TkDodo  do you suggest we make those changes in a separate PR or should I add annotations on this PR itself?

Reply:
I see 3 ways forward here:

1) you can checkout the PR with the gh cli and work on it directly
2) you can create a PR _to_ this PR :)
3) we can merge this to v5, as it will not be released, and you can work on it in a separate issue / PR

whatever you prefer is fine with me :)

Reply:
I vote for merging it if it's good enough for now.
This PR is massive and it would be hard to keep up to date, when other PRs start merging to v5.

Reply:
yes, you're right @DamianOsipiuk . There are already conflicts, sorry for that. Once they are fixed, I'll merge this PR next

Reply:
Yeah, I agree! The scope for the other `solid-query` v5 changes can be addressed in a separate PR

Reply:
the merge conflicts are fixed now 🫡

Reply:
@TkDodo merge conflicts are fixed 🫡